src/hash.ts

```diff
@@ -148,3 +115,39 @@ export async function initHasher(): Promise<unknown> {
   }
   return hasherPromise;
 }
+
+let tempHashCounter = 0;
+
+const tempPrefix = '/t/';
+
+export function newTempHash(): Hash {
+  // Must not overlap with Hash.prototype.toString results
+  return (tempPrefix +
+    (tempHashCounter++)
+      .toString()
+      .padStart(HASH_LENGTH - tempPrefix.length, '0')) as unknown as Hash;
+}
+
+export function isHash(v: unknown): v is Hash {
+  return typeof v === 'string' && hashRe.test(v);
+}
+
+export function isTempHash(
+  hash: Hash,
+): hash is `/t/${string}` & {[hashTag]: true} {
+  return (hash as unknown as string).startsWith('/t/');
+}
+
+export function assertNotTempHash<H extends Hash>(
+  hash: H,
+): asserts hash is H extends `/t/${string}` ? never : H {
```

Comment:
I need to study my advanced typescript :)

Reply:
Not sure this is worth doing.

Reply:
I removed this. This now just returns `void`